### PR TITLE
feat(eventcatalog): rendering of asyncapi files in service page

### PR DESCRIPTION
**Pull request by lcsbltm**
## Motivation

The possibility of displaying a Swagger UI based on OpenAPI spec files in a Service page is great, however, due to the nature of the API, sometimes it is necessary to write the documentation using a different spec like AsyncAPI.

This feature looks for `async` files in the services folders and uses https://github.com/asyncapi/asyncapi-react to render the documentation in a new MDX component `<AsyncAPI />`.

Attached is an example of an AsyncAPI file displayed in the Service page:

Animation

### Have you read the [Contributing Guidelines on pull requests](https://github.com/boyney123/eventcatalog/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### CODE REVIEW

**Comment by dpwdec** [comment]
Wow! Excellent work. 

This is great for those of us that use AsyncAPI as our primary source of truth for documenting messages. 

Thank you for doing this! 🤩

**Comment by boyney123** [comment]
Hey @lcsbltm 

Loving this! Looks great! Need some time to review it, but looking awesome! Will try and get this in ASAP.

File: packages/eventcatalog/lib/file-reader.ts [hunk 1]
```diff
@@ -56,6 +56,19 @@ export const getOpenAPISpecFromDir = (pathToOpenAPISpecDir: string): string => {
   }
 };
 
+export const getAsyncAPISpecFromDir = (pathToAsyncAPISpecDir: string): string => {
+  try {
+    const files = fs.readdirSync(pathToAsyncAPISpecDir);
+
+    // See if any oas are in there, ignoring extension
+    const openAPIFileName = files.find((fileName) => fileName.includes('asyncapi'));
```
Comment (boyney123) [inline packages/eventcatalog/lib/file-reader.ts:64]:
```suggestion
    const asyncAPIFile = files.find((fileName) => fileName.includes('asyncapi'));
```

File: packages/eventcatalog/lib/file-reader.ts [hunk 2]
```diff
@@ -56,6 +56,19 @@ export const getOpenAPISpecFromDir = (pathToOpenAPISpecDir: string): string => {
...
+    if (!openAPIFileName) throw new Error('No schema found');
```
Comment (boyney123) [inline packages/eventcatalog/lib/file-reader.ts:65]:
```suggestion
    if (!asyncAPIFile) throw new Error('No async api file found');
```

File: packages/eventcatalog/lib/file-reader.ts [hunk 3]
```diff
@@ -56,6 +56,19 @@ export const getOpenAPISpecFromDir = (pathToOpenAPISpecDir: string): string => {
...
+    return fs.readFileSync(path.join(pathToAsyncAPISpecDir, openAPIFileName), 'utf-8');
```
Comment (boyney123) [inline packages/eventcatalog/lib/file-reader.ts:66]:
```suggestion
    return fs.readFileSync(path.join(pathToAsyncAPISpecDir, asyncAPIFile), 'utf-8');
```

**Comment by boyney123** [comment]
Few comments in the code review, but just tested and everything working great!

I think it would be good for us in the future to support multiple files, but let's go with this for now, as it's a great start! And then we can see where community/requests come from in the future.

**Comment by boyney123** [comment]
Also the CI is failing, if you could try and lint the project that would be great, any questions just ask

**Comment by lcsbltm** [comment]
Thanks for the review and comments guys. 

@boyney123 - I committed the changes based on your comment, and I'll try to fix the CI error. I'll ping you if I need help 😃

**Comment by lcsbltm** [comment]
Hey @boyney123, I gave it a try fixing the CI error. I hope it the checks pass. I think the problem was that the `yarn.lock` file was not immutable. From my local tests, it is immutable now.

**Comment by boyney123** [comment]
Great thanks @lcsbltm for taking a look, I will give it another go and have a look!

**Comment by boyney123** [comment]
@all-contributors add @lcsbltm for code

**Comment by boyney123** [comment]
Thank you @lcsbltm for the contribution! Great addition to EventCatalog, I love it.


Added you on the contributor list and also you can find your release here: https://github.com/boyney123/eventcatalog/releases

To get it working

Update the following packages

`[@eventcatalog/types@0.3.0](https://github.com/boyney123/eventcatalog/releases/tag/%40eventcatalog%2Ftypes%400.3.0)`
https://github.com/boyney123/eventcatalog/releases/tag/%40eventcatalog%2Fcore%400.4.0

**Comment by s1mrankaur** [comment]
@boyney123  I use https://www.npmjs.com/package/@eventcatalog/plugin-doc-generator-asyncapi to generate eventcatalog. I tried to add <AsyncAPI /> to the event.md file and got

`Component AsyncAPI was not imported, exported, or provided by MDXProvider as global scope`

I am using version 3 of AsyncAPI.

**Pull request merged**